linux-manual: make kernel-doc.py and split-man.pl executable so patchShebangs doesn't skip them

fcodes0 opened the pull request:
make kernel-doc.py and split-man.pl executable so patchShebangs doesn't skip them.

This fixes the `kernel-doc: scripts/kernel-doc: /usr/bin/env: bad interpreter: No such file or directory` error allowing the package to build successfully.
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

emilazy committed:
nixos/installation-device: don’t include `stdenv`

A basic NixOS system shouldn’t require `stdenv` to install, and
it adds something like 45 MiB of build tools to the installer
closure. This doesn’t really help much at present since
WebKitGTK(?!) has a dodgy reference to GCC, but it’s a step in the
right direction.

Bert-Proesmans committed:
nixos/kanidm: bind ca-bundle to validate url on provisioning

The provisioning script can be configured to validate the server certificate,
but the unit lockdown prevents access to the trusted certificate authority
(ca) bundle.
The system trusted ca store path is added to the bind paths as solution.

frederictobiasc committed:
nixos/traccar: Rework configuration file creation.

Previously, a flat attribute set was expected. The new implementation
uses nested attribute sets. This is a breaking change.

RossSmyth committed:
isle-portable: init at 0-unstable-2025-07-25

LordGrimmauld committed:
nixos/tests/audit: test plugins

r-ryantm committed:
opensubdiv: 3.6.0 -> 3.6.1

r-ryantm committed:
libdatachannel: 0.23.0 -> 0.23.1

r-ryantm committed:
spdx-license-list-data: 3.26.0 -> 3.27.0

r-ryantm committed:
osl: 1.14.5.1 -> 1.14.6.0

r-ryantm committed:
mdk-sdk: 0.33.0 -> 0.33.1

r-ryantm committed:
redwax-tool: 0.9.9 -> 1.0.0

r-ryantm committed:
marble-shell-theme: 48.0.1 -> 48.3.2

r-ryantm committed:
libsolv: 0.7.33 -> 0.7.34

r-ryantm committed:
bicep: 0.36.1 -> 0.36.177

r-ryantm committed:
tkdiff: 5.7 -> 6.0

r-ryantm committed:
zxtune: 5100 -> 5101

r-ryantm committed:
ios-webkit-debug-proxy: 1.9.1 -> 1.9.2

r-ryantm committed:
hubstaff: 1.6.31-a6da06ad -> 1.7.3-6c31e21a

r-ryantm committed:
acr-cli: 0.15 -> 0.16

r-ryantm committed:
gmm: 5.4.2 -> 5.4.4

r-ryantm committed:
prometheus-frr-exporter: 1.7.0 -> 1.8.0

r-ryantm committed:
vrpn: 07.35 -> 07.36

r-ryantm committed:
ipatool: 2.1.6 -> 2.2.0

r-ryantm committed:
scotch: 7.0.7 -> 7.0.8

r-ryantm committed:
gnu-shepherd: 1.0.5 -> 1.0.6

r-ryantm committed:
localstack: 4.5.0 -> 4.6.0

r-ryantm committed:
apt: 3.1.2 -> 3.1.3

r-ryantm committed:
pgmoneta: 0.17.2 -> 0.18.0

r-ryantm committed:
tarlz: 0.28 -> 0.28.1

r-ryantm committed:
junicode: 2.211 -> 2.216

r-ryantm committed:
google-alloydb-auth-proxy: 1.13.3 -> 1.13.4

r-ryantm committed:
rocksdb: 10.2.1 -> 10.4.2

r-ryantm committed:
checkstyle: 10.25.1 -> 10.26.1

r-ryantm committed:
amneziawg-tools: 1.0.20241018 -> 1.0.20250706

r-ryantm committed:
phoenixd: 0.6.0 -> 0.6.1

r-ryantm committed:
micronaut: 4.9.0 -> 4.9.1

r-ryantm committed:
numcpp: 2.14.1 -> 2.14.2

r-ryantm committed:
youtrack: 2025.1.82518 -> 2025.1.86199

r-ryantm committed:
mockoon: 9.2.0 -> 9.3.0

r-ryantm committed:
flrig: 2.0.07 -> 2.0.08

r-ryantm committed:
rocketchat-desktop: 4.7.0 -> 4.7.1

r-ryantm committed:
fzy: 1.0 -> 1.1

r-ryantm committed:
morgen: 3.6.15 -> 3.6.16

r-ryantm committed:
doomretro: 5.7 -> 5.7.1

r-ryantm committed:
linuxPackages.universal-pidff: 0.1.1 -> 0.2.0

r-ryantm committed:
linuxPackages.akvcam: 1.2.7 -> 1.3.0

r-ryantm committed:
uxn: 1.0-unstable-2025-07-03 -> 1.0-unstable-2025-07-12

r-ryantm committed:
mopidy-tidal: 0.3.9 -> 0.3.10

r-ryantm committed:
gopeed: 1.7.0 -> 1.7.1

r-ryantm committed:
bootstrap-studio: 7.1.1 -> 7.1.2

r-ryantm committed:
solo5: 0.9.1 -> 0.9.2

r-ryantm committed:
protozero: 1.8.0 -> 1.8.1

r-ryantm committed:
traccar: 6.7.3 -> 6.8.1

r-ryantm committed:
commonsLang: 3.17.0 -> 3.18.0

r-ryantm committed:
dpkg: 1.22.20 -> 1.22.21

r-ryantm committed:
boxflat: 1.33.0 -> 1.34.2

r-ryantm committed:
cadaver: 0.26 -> 0.27

r-ryantm committed:
samtools: 1.21 -> 1.22.1

r-ryantm committed:
freeciv: 3.1.5 -> 3.2.0

r-ryantm committed:
converseen: 0.14.0.0 -> 0.15.0.2

r-ryantm committed:
graalvmPackages.graalvm-ce: 24.0.1 -> 24.0.2

womeier committed:
coqPackages.wasmcert: init at 2.1.0

Co-authored-by: Pierre Roux <pierre.roux@onera.fr>

r-ryantm committed:
wxsqlite3: 4.10.9 -> 4.10.11

sarahec committed:
langgraph-cli: 0.2.10 -> 0.3.6

https://github.com/langchain-ai/langgraph/releases/tag/cli%3D%3D0.3.6

r-ryantm committed:
sby: 0.52 -> 0.55

r-ryantm committed:
sysdig-cli-scanner: 1.22.4 -> 1.22.5

r-ryantm committed:
gauge-unwrapped: 1.6.18 -> 1.6.19

r-ryantm committed:
tartube-yt-dlp: 2.5.100 -> 2.5.156

r-ryantm committed:
capacities: 1.48.7 -> 1.50.4

r-ryantm committed:
svelte-language-server: 0.17.16 -> 0.17.17

r-ryantm committed:
cloudlog: 2.6.19 -> 2.6.20

r-ryantm committed:
hevea: 2.36 -> 2.37

r-ryantm committed:
qtractor: 1.5.6 -> 1.5.7

r-ryantm committed:
dracula-theme: 4.0.0-unstable-2025-06-11 -> 4.0.0-unstable-2025-07-17

r-ryantm committed:
marp-cli: 4.2.0 -> 4.2.1

r-ryantm committed:
tinty: 0.27.0 -> 0.28.0

siraben committed:
python3Packages.eth-typing: 5.1.0 -> 5.2.1

siraben committed:
python3Packages.eth-utils: 5.1.0 -> 5.3.0

r-ryantm committed:
hamrs-pro: 2.41.1 -> 2.42.1

r-ryantm committed:
txr: 301 -> 302

r-ryantm committed:
blobfuse: 2.4.2 -> 2.5.0

r-ryantm committed:
ibmcloud-cli: 2.34.2 -> 2.35.0

siraben committed:
python3Packages.eth-account: 0.13.5 -> 0.13.7

r-ryantm committed:
brewtarget: 4.1.2 -> 4.1.3

r-ryantm committed:
iosevka-bin: 33.2.6 -> 33.2.7

r-ryantm committed:
mercure: 0.19.2 -> 0.20.0

vbgl committed:
ocamlPackages.ocsipersist: 1.1.0 → 2.0.0

vbgl committed:
ocamlPackages.ocsipersist-sqlite-config: init at 2.0.0

ocamlPackages.ocsipersist-pgsql-config: init at 2.0.0

thtrf committed:
wails: move to by-name

thtrf committed:
wails: 2.10.1 -> 2.10.2

r-ryantm committed:
papilo: 2.4.2 -> 2.4.3

vbgl committed:
ocamlPackages.ocsigen-start: 6.2.0 → 7.1.0

siraben committed:
python3Pacakges.py-ecc: 7.0.0 -> 8.0.0

siraben committed:
python3Packages.rlp: 4.0.0 -> 4.1.0

siraben committed:
python3Packages.trie: 3.0.1 -> 3.1.0

liberodark committed:
pff-tools: init at 0-unstable-2025-07-22

r-ryantm committed:
free42: 3.3.6 -> 3.3.8

r-ryantm committed:
upcloud-cli: 3.20.2 -> 3.21.0

r-ryantm committed:
angie: 1.9.1 -> 1.10.1

siraben committed:
python3Packages.eth-keyfile: 0.8.1 -> 0.9.1

siraben committed:
python3Packages.eth-keys: 0.6.0 -> 0.7.0

siraben committed:
python3Packages.hexbytes: 1.2.0 -> 1.3.1

siraben committed:
python3Packages.py-evm: 0.10.1-beta.2 -> 0.12.1-beta.1

siraben committed:
python3Packages.eth-abi: 5.1.0 -> 5.2.0

siraben committed:
python3Packages.eth-tester: 0.12.0-beta.2 -> 0.13.0-beta.1

siraben committed:
python3Packages.eth-rlp: 2.1.0 -> 2.2.0

siraben committed:
python3Packages.web3: 7.8.0 -> 7.12.1

siraben committed:
python3Packages.slither-analyzer: 0.10.3 -> 0.11.3

r-ryantm committed:
schemacrawler: 16.26.2 -> 16.26.3

r-ryantm committed:
redis-plus-plus: 1.3.14 -> 1.3.15

petrzjunior committed:
kssd: 2.21 -> 2.21-unstable-2024-05-27

r-ryantm committed:
diamond: 2.1.12 -> 2.1.13

r-ryantm committed:
dbmate: 2.27.0 -> 2.28.0

r-ryantm committed:
youtubeuploader: 1.25.1 -> 1.25.3

r-ryantm committed:
kn: 1.18.0 -> 1.19.0

r-ryantm committed:
grpc_cli: 1.73.0 -> 1.74.0

r-ryantm committed:
exoscale-cli: 1.85.3 -> 1.85.4

r-ryantm committed:
rat-king-adventure: 2.2.2 -> 2.2.3

r-ryantm committed:
kid3: 3.9.6 -> 3.9.7

r-ryantm committed:
elasticmq-server-bin: 1.6.12 -> 1.6.14

r-ryantm committed:
autobrr: 1.63.1 -> 1.64.0

r-ryantm committed:
jhentai: 8.0.8+295 -> 8.0.9

r-ryantm committed:
cherry-studio: 1.4.8 -> 1.5.3

r-ryantm committed:
nghttp3: 1.10.1 -> 1.11.0

r-ryantm committed:
suitesparse-graphblas: 10.1.0 -> 10.1.1

r-ryantm committed:
cppcheck: 2.17.1 -> 2.18.0

r-ryantm committed:
sea-orm-cli: 1.1.12 -> 1.1.14

r-ryantm committed:
redux: 1.3.5 -> 1.4.1

r-ryantm committed:
jay: 1.10.0 -> 1.11.0

r-ryantm committed:
watchyourlan: 2.1.2 -> 2.1.3

bryanhonof committed:
python3Packages.mcp: 1.9.4 -> 1.12.2

I also had to disable some failing tests that I didn't mananage to catch in a
reproducible manner. They tend to happen when the system I'm running
nixpkgs-review on is under some load, but they don't happen all that often when
building the package in isolation.

Co-authored-by: Joshua Peek <josh@users.noreply.github.com>

bryanhonof committed:
python3Packages.openai-agents: 0.0.13 -> 0.2.3

Got rid of nix-update-script, since it didn't seem to be doing anything?

ryota2357 committed:
koka: 3.1.2 -> 3.2.2

ryota2357 committed:
koka: migrate to pkgs/by-name

ryota2357 committed:
koka: update executableHaskellDepends

r-ryantm committed:
trealla: 2.78.24 -> 2.79.2

bryanhonof committed:
python3Packages.mcp: add bryanhonof as maintainer

bryanhonof committed:
python3Packages.fastmcp: 2.8.0 -> 2.10.6

bryanhonof committed:
python3Packages.fastapi-mcp: 0.3.4 -> 0.3.7

ethancedwards8 committed:
python3Packages.xcaplib: init at 2.0.1-unstable-2025-03-20

NGI project.

Co-authored-by: Acid Bong <acidbong@tilde.club>
Signed-off-by: Ethan Carter Edwards <ethan@ethancedwards.com>

ethancedwards8 committed:
python3Packages.msrplib: init at 0.20.1-unstable-2021-06-01

NGI project.

Signed-off-by: Ethan Carter Edwards <ethan@ethancedwards.com>

4evy committed:
koboldcpp: 1.96 -> 1.96.2

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.96.2
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.96...v1.96.2

4evy committed:
koboldcpp: remove `env.NIX_C{XX,}FLAGS_COMPILE`

r-ryantm committed:
zpaqfranz: 62.2 -> 62.4

r-ryantm committed:
seamly2d: 2025.6.23.216 -> 2025.7.21.216

r-ryantm committed:
ark-pixel-font: 2025.03.14 -> 2025.07.21

r-ryantm committed:
clementine: 1.4.1-45-g34eb666c0 -> 1.4.1-48-g12e851937

r-ryantm committed:
krillinai: 1.3.0 -> 1.3.1

bryanhonof committed:
python3Packages.gradio: 5.38.0 -> 5.38.2

r-ryantm committed:
jmol: 16.3.27 -> 16.3.31

bryanhonof committed:
oterm: allow newer versions of fastmcp dependency

We package a newer version of fastmcp already, but upstream seems to be limiting
it to 2.9.

r-ryantm committed:
bluemap: 5.8 -> 5.10

r-ryantm committed:
seq66: 0.99.20 -> 0.99.21

bryanhonof committed:
mcp-proxy: disable tests that are failing

These started popping up when bumping the python3Packages.mcp version.

bryanhonof committed:
python3Packages.smolagents: disable failing tests

r-ryantm committed:
flow: 0.274.2 -> 0.277.1

Defelo committed:
psitransfer: use finalAttrs pattern

Defelo committed:
psitransfer: use tag in fetchFromGitHub

Defelo committed:
psitransfer: add meta.changelog

Defelo committed:
psitransfer: add updateScript

Defelo committed:
psitransfer: 2.2.0 -> 2.3.0

Changelog: https://github.com/psi-4ward/psitransfer/releases/tag/v2.3.0
Diff: https://github.com/psi-4ward/psitransfer/compare/v2.2.0...v2.3.0

r-ryantm committed:
ocamlPackages.fmt: 0.10.0 -> 0.11.0

r-ryantm committed:
gwyddion: 2.68 -> 2.69

r-ryantm committed:
dart: 3.8.1 -> 3.8.2

r-ryantm committed:
readsb: 3.14.1683 -> 3.14.1691

r-ryantm committed:
lynis: 3.1.4 -> 3.1.5

r-ryantm committed:
spider: 2.37.120 -> 2.37.136

r-ryantm committed:
discordo: 0-unstable-2025-06-26 -> 0-unstable-2025-07-28

r-ryantm committed:
docker-buildx: 0.25.0 -> 0.26.1

r-ryantm committed:
whisparr: 2.0.0.1112 -> 2.0.0.1171

r-ryantm committed:
plasticity: 25.1.9 -> 25.2.2

r-ryantm committed:
twilio-cli: 6.0.1 -> 6.1.0

r-ryantm committed:
amazon-cloudwatch-agent: 1.300057.1 -> 1.300058.1

proux01 committed:
coqPackages.interval: 4.11.2 -> 4.11.3

r-ryantm committed:
bazel-watcher: 0.26.3 -> 0.26.8

r-ryantm committed:
opentelemetry-collector-builder: 0.129.0 -> 0.131.0

r-ryantm committed:
youtube-tui: 0.8.3 -> 0.9.0

bryanhonof committed:
home-assistant: ignore Pydantic deprecation warnigns for mcp package

We bumped the python3Packages.mcp package, and were able to remove this set of
flags due to the Pydantic library also being updated. However, since
home-assistant still seems to be using 1.5.0, and overriding it here, these
flags are still needed in this particular case.

r-ryantm committed:
buf: 1.55.1 -> 1.56.0

r-ryantm committed:
albyhub: 1.18.2 -> 1.18.5

r-ryantm committed:
amp-cli: 0.0.1752566512-ga67426 -> 0.0.1754036264-ga0aa13

r-ryantm committed:
vue-language-server: 3.0.1 -> 3.0.4

r-ryantm committed:
yandex-cloud: 0.153.0 -> 0.156.0

stepbrobd committed:
ocamlPackages.js_of_ocaml: 6.1.1 -> 6.2.0

keegancsmith committed:
amp-cli: update entrypoint

r-ryantm committed:
luanti-client: 5.12.0 -> 5.13.0

49016 committed:
lib.modules: Add hint when using `config` in `imports`

r-ryantm committed:
zoekt: 3.7.2-2-unstable-2025-06-27 -> 3.7.2-2-unstable-2025-07-30

r-ryantm committed:
ecs-agent: 1.96.0 -> 1.97.0

r-ryantm committed:
tetragon: 1.4.0 -> 1.5.0

Prince213 committed:
luau: 0.684 -> 0.685

r-ryantm committed:
tutanota-desktop: 293.250630.0 -> 299.250725.1

r-ryantm committed:
ultrastardx: 2025.6.0 -> 2025.8.0

roberth committed:
lib.modules: Test infinite recursion hint

We don't want it to occur in the trace of any unrelated errors.

roberth committed:
lib.modules: Generalize the import hint to _module.args

roberth committed:
lib.modules: Make _module.args evaluation explicit in trace

roberth committed:
lib.modules: Adjust error message

- Lower case error trace for consistency
- Be more explicit about the condition under which the hint applies,
  and the resolution.

amadejkastelic committed:
freeplane: 1.12.10 -> 1.12.12

Gliczy committed:
crispy-doom: refactor

Gliczy committed:
crispy-doom: add missing dependencies

Gliczy committed:
crispy-doom: add optional truecolor support

Gliczy committed:
crispy-doom: add myself as maintainer

Gliczy committed:
crispy-doom: add keenanweaver as maintainer

r-ryantm committed:
benthos: 4.53.0 -> 4.54.0

r-ryantm committed:
crun: 1.22 -> 1.23.1

r-ryantm committed:
trexio: 2.5.0 -> 2.6.0

r-ryantm committed:
json-fortran: 9.0.3 -> 9.0.4

r-ryantm committed:
odpic: 5.6.0 -> 5.6.2

r-ryantm committed:
ast-grep: 0.39.1 -> 0.39.2

r-ryantm committed:
wtfutil: 0.44.1 -> 0.45.0

r-ryantm committed:
strictdoc: 0.9.3 -> 0.10.1

fabaff committed:
trivy: 0.64.1 -> 0.65.0

Diff: https://github.com/aquasecurity/trivy/compare/refs/tags/v0.64.1...refs/tags/v0.65.0

Changelog: https://github.com/aquasecurity/trivy/releases/tag/v0.65.0

fabaff committed:
vunnel: 0.35.1 -> 0.36.0

Diff: https://github.com/anchore/vunnel/compare/refs/tags/v0.35.1...refs/tags/v0.36.0

Changelog: https://github.com/anchore/vunnel/releases/tag/v0.36.0

amadejkastelic committed:
linuxPackages.yt6801: 1.0.29-20240812 -> 1.0.30-20250430

r-ryantm committed:
omnictl: 0.52.0 -> 1.0.0

amadejkastelic committed:
linuxPackages.yt6801: fix build for 6.16

LordGrimmauld committed:
nixos/auditd: support configuring service through nix

LordGrimmauld committed:
nixos/tests/audit: add grimmauld to maintainers

r-ryantm committed:
python3Packages.disposable-email-domains: 0.0.128 -> 0.0.129

x123 committed:
opencode: 0.3.112 -> 0.3.122

mattchrist committed:
nixos/freshrss: fix loading extensions' static content

Before this change, the THIRDPARTY_EXTENSIONS_PATH would end up with a
double-slash in the path, which was breaking FreshRSS's is_valid_path
detection.

r-ryantm committed:
jawiki-all-titles-in-ns0: 0-unstable-2025-07-01 -> 0-unstable-2025-08-01

illdefined reviewed (commented):
That is some nasty behaviour of `patchShebangs`. It may make sense to skip non‐executable files when recursively patching a directory tree, but perhaps less so for individual files explicitly provided on the command line.

I am surprised that nobody noticed before, but perhaps it’s rare for build environments to lack `/usr/bin/env`.

I think that your PR is acceptable as is, but using a list would avoid repetition and make it easier to expand in the future.

ddogfoodd committed:
typstyle: 0.13.16 -> 0.13.17

Changelog: https://github.com/typstyle-rs/typstyle/releases/tag/v0.13.17
Diff:
https://github.com/typstyle-rs/typstyle/compare/v0.13.16...v0.13.17

Aleksanaa committed:
gnome-commander: fix build by temporarily disable gtest

fcodes0 commented on pkgs/by-name/li/linux-manual/package.nix:
```diff
@@ -25,6 +25,7 @@ stdenv.mkDerivation {
     # Use scripts/kernel-doc.py here, not scripts/kernel-doc because
     # patchShebangs skips symlinks
 
+    chmod +x scripts/kernel-doc.py scripts/split-man.pl
```
Thanks for the feedback! This suggestion makes sense, I'll apply those changes.

OPNA2608 committed:
re-isearch: 2.20220925.4.0a-unstable-2025-03-16 -> 2.20220925.4.0a-unstable-2025-05-15

OPNA2608 committed:
steamworks: Various fixes

- Change updateScript to gitUpdater, nix-update-script was trying to downgrade to the last tag with a release page
- Add catch2 to checkInputs, was previously unused
- Fix catch2 linking
- Fix non-debug compilation
- Enable tests
- Add references to upstream-submitted fixes

OPNA2608 committed:
re-isearch: Disable enableParallelBuilding again

Do so explicitly, and add a note about why.

7c6f434c committed:
tptp: 7.2.0 -> 9.1.0

r-ryantm committed:
picocrypt-cli: 1.48 -> 1.49

r-ryantm committed:
typescript-go: 0-unstable-2025-07-25 -> 0-unstable-2025-08-01

r-ryantm committed:
render50: 9.2.8 -> 9.2.10

illdefined commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 430657`
Commit: `83612371fd8faf318b42ac52d41940c50d0d3d72`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>linux-manual</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>linux-manual</li>
  </ul>
</details>

flokli committed:
libdigidoc: add myself as maintainer

I'd like to get notifications for bumps of this, considering it's used
in qdigidoc, a package I maintain.

r-ryantm committed:
gallery-dl: 1.30.0 -> 1.30.2

LordGrimmauld committed:
nixos/auditd: add plugin support

LordGrimmauld committed:
nixos/auditd: add built-in plugins (default disabled)

illdefined commented:
It might make sense to squash the commits, but I have no strong preference.

Another nitpick: The commit titles could be made a bit more concise and use imperative voice (second commit), with additional information in the body:

> linux-manual: make scripts executable for patchShebangs
> 
> patchShebangs ignores non‐executable files, resulting in build errors if /usr/bin/env is unavailable.


> linux-manual: provide scripts to patch as list
> 
> This avoids repetition and permits easier expansion.

ddogfoodd committed:
tinymist: 0.13.16 -> 0.13.18

Changelog:
https://github.com/Myriad-Dreamin/tinymist/releases/tag/v0.13.18
Diff:
https://github.com/Myriad-Dreamin/tinymist/compare/v0.13.16...v0.13.18

fabaff committed:
python313Packages.cxxheaderparser: 1.5.0 -> 1.5.4

Diff: https://github.com/robotpy/cxxheaderparser/compare/1.5.0...1.5.4

r-ryantm committed:
hyprutils: 0.8.1 -> 0.8.2

GaetanLepage committed:
presenterm: 0.15.0 -> 0.15.1

Diff: https://github.com/mfontanini/presenterm/compare/refs/tags/v0.15.0...refs/tags/v0.15.1

Changelog: https://github.com/mfontanini/presenterm/releases/tag/v0.15.1

r-ryantm committed:
pimsync: 0.4.3 -> 0.4.4

4evy committed:
raycast: 1.102.0 -> 1.102.3

thtrf committed:
satisfactorymodmanager: use webkitgtk2-4.1

4evy committed:
warp-terminal: 0.2025.07.23.08.12.stable_02 -> 0.2025.07.30.08.12.stable_02

Changelog: https://docs.warp.dev/getting-started/changelog#id-2025.07.30-v0.2025.07.30.08.12

r-ryantm committed:
mint-themes: 2.3.0 -> 2.3.1

r-ryantm committed:
codebook: 0.3.5 -> 0.3.6

r-ryantm committed:
shogihome: 1.24.2 -> 1.24.3

jfly commented on pkgs/by-name/li/linux-manual/package.nix:
```diff
@@ -21,14 +21,19 @@ stdenv.mkDerivation {
-  '';
+  postPatch =
+    let
+      scripts = lib.escapeShellArgs [
```
What are you suggesting instead? `lib.concatStringsSep " "`? As this code is structured, we need a nix string here, not a list, because we're going to interpolate it into a nix string.

Even thought `lib.concatStringsSep` would work here, I'd prefer `lib.escapeShellArgs`. Imagine if someone were to add a file in the future with a space in it.

illdefined replied:
`toString` could be used as it is `lib.concatStringsSep " "`. I believe however that this is bad practice, even if it is very common in nixpkgs.

The _current_ list definition does not contain any strings with characters requiring escaping, it is however used as an argument for a shell command and as such should always be escaped defensively in my opinion.

jfly commented on pkgs/by-name/li/linux-manual/package.nix:
```diff
@@ -21,14 +21,19 @@ stdenv.mkDerivation {
+  postPatch =
+    let
+      scripts = lib.escapeShellArgs [
+        # Use scripts/kernel.py rather than scripts/kernel.doc because
```
I don't understand this comment. I don't see `scripts/kernel.py` nor `scripts/kernel.doc` mentioned anywhere in this diff.

jfly reviewed (approved):
Thanks! This fixes the build of `pkgs.linuxPackages_latest` (Linux 6.16) for me.

> That is some nasty behaviour of patchShebangs. It may make sense to skip non‐executable files when recursively patching a directory tree, but perhaps less so for individual files explicitly provided on the command line.

+1 to this

Also, do we know if anyone has tried to upstream the executable bits on these scripts?

r-ryantm committed:
LycheeSlicer: 7.4.0 -> 7.4.2

liberodark committed:
sfizz: 1.2.1 -> 1.2.3

liberodark committed:
tpnotes: switch to finalAttrs

roberth committed:
lib.modules: Improve infinite recursion hint

hsjobeki: Using config in imports is possible in general. But its not possible to do conditional imports where the condition depends on config. Thats two different statements.

Co-authored-by: Johannes Kirschbauer <hsjobeki+github@gmail.com>
